app/com/lynxanalytics/biggraph/graph_operations/ExportToNeo4j.scala

```diff
@@ -0,0 +1,135 @@
+package com.lynxanalytics.biggraph.graph_operations
+
+import com.lynxanalytics.biggraph.graph_api._
+import com.lynxanalytics.biggraph.graph_util.HadoopFile
+import org.apache.spark
+import org.apache.spark.sql.SaveMode
+
+object ExportAttributesToNeo4j extends OpFromJson {
+  class Input extends MagicInputSignature {
+    val t = table
+  }
+  class Output(implicit instance: MetaGraphOperationInstance) extends MagicOutput(instance) {
+    val exportResult = scalar[String]
+  }
+  def fromJson(j: JsValue) = ExportAttributesToNeo4j(
+    (j \ "url").as[String], (j \ "username").as[String], (j \ "password").as[String],
+    (j \ "labels").as[String], (j \ "keys").as[Seq[String]], (j \ "version").as[Long],
+    (j \ "nodesOrRelationships").as[String])
+}
+
+case class Neo4jConnection(url: String, username: String, password: String) {
+  def send(df: spark.sql.DataFrame, query: String) {
+    df.write
+      .format("org.neo4j.spark.DataSource")
+      .option("authentication.type", "basic")
+      .option("authentication.basic.username", username)
+      .option("authentication.basic.password", password)
+      .option("url", url)
+      .option("query", query)
+      .save()
+  }
+}
+
+case class ExportAttributesToNeo4j(
+    url: String, username: String, password: String, labels: String, keys: Seq[String],
+    version: Long, nodesOrRelationships: String)
+  extends SparkOperation[ExportAttributesToNeo4j.Input, ExportAttributesToNeo4j.Output] {
+  val neo = Neo4jConnection(url, username, password)
```

Comment:
When I first saw this (and of course didn't realize I just reviewed the definition of this class :( ) I thought why would you put a connection kind of class up here, why not create the connection only when needed, in execute. But of course, there is no connection being made here, it just collects some metadata which will be used in a helper method of this object. So then what about naming it differently? E.g. Neo4JHelper?

Reply:
> When I first saw this (and of course didn't realize I just reviewed the definition of this class :( ) I thought why would you put a connection kind of class up here, why not create the connection only when needed, in execute. But of course, there is no connection being made here, it just collects some metadata which will be used in a helper method of this object. So then what about naming it differently? E.g. Neo4JHelper?

I changed it to Neo4jConnectionParameters.